### PR TITLE
[FIX] use provided name for firstname and lastname 

### PR DESCRIPTION
**Pull request by wpichler**
on copy if no firsrname and lastname is specified

### CODE REVIEW

**Comment by wpichler** [comment]
This PR does fix this problem:
With the partner_firstname module installed - when you do edit your template user - then you have to specify firstname and lastname to be able to save it. Then someone new does signup on your website - on signup the copy user function does get called. On copy name is set in defaults - but not firstname and lastname - so the copy function will use firstname and lastname from the template user - and will overwrite the provided name.

**Comment by gdgellatly** [comment]
@wpichler Hi, can you write a test for this improvement please.  The travis failure seems unrelated.  I copied your code to #648 so will add the test there as well.

**Comment by wpichler** [comment]
@gdgellatly I've now written a test - and found a bug with it ;-) But this is my first test case i every wrote for an odoo module (shame on me...) - so please take a look at, and check if it looks correct...

**Comment by gdgellatly** [comment]
@wpichler what is the bug you found?  

I'm not the best on tests, but I've had to spend some time with the tests in this module.  The tests ultimately inherit from `tests/base.py` but I guess you worked that out.  There is lots of helpers in there, but this module has quite complicated tests compared to most.

I'm wondering if at least for some of this, setting the copy context is all that is required rather than manually setting the string, but I can see you've checked upstream so I won't second guess it.

**Comment by wpichler** [comment]
@gdgellatly I did not imported _ ....
About the tests - I've implemented them - and they seem to work - and i found a bug with them - so they tests did their work quit good ;-)

**Comment by gdgellatly** [comment]
@wpichler alright, fix these 2 lint errors and you'll be good for review

======== Testing test_flake8 ========
partner_firstname/models/res_users.py:43:80: E501 line too long (96 > 79 characters)
partner_firstname/models/res_users.py:44:80: E501 line too long (93 > 79 characters)

**Comment by wpichler** [comment]
Fixed lint errors

File: partner_firstname/tests/test_copy.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def create_original(self):
+        self.original = self.env["res.users"].create({
+            "firstname": u"Firstname",
+            "lastname": u"Lastname",
+            "name": u"Firstname Lastname",
+            "login": u"firstname.lastname"
+        })
+
+    def tearDown(self):
```
Comment (NL66278) [inline partner_firstname/tests/test_copy.py]: No need for this method, as you do not add to the super method.

File: partner_firstname/tests/test_copy.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        super(UserCase, self).tearDown()
+
+    def compare(self, copy):
+        self.assertEqual(copy.lastname, u"Lastname2")
+        self.assertEqual(copy.firstname, u"Firstname2")
+        self.assertEqual(copy.name, u"Lastname2 Firstname2")
+
+    def test_copy_name(self):
+        """Copy original with default name set - firstname lastname not set."""
+        copy = self.original.copy({
```
Comment (NL66278) [inline partner_firstname/tests/test_copy.py]: In your copy methods use the default= keyword. It works without, but it is clearer to call a keyword argument with a keyword.

**Review by NL66278 — CHANGES_REQUESTED** [review]
See my other comments

File: partner_firstname/tests/test_copy.py (hunk 2 above)
Comment (wpichler) [reply partner_firstname/tests/test_copy.py]: done

File: partner_firstname/tests/test_copy.py (hunk 1 above)
Comment (wpichler) [reply partner_firstname/tests/test_copy.py]: done

**Comment by rafaelbn** [comment]
Dear @wpichler , Thanks for this contribution. Could you please attend comments? Thanks! Travis is red :red_circle:

**Comment by wpichler** [comment]
fixed the bug in the copy test case (typo) - and rebased it against origin/10.0

**Comment by wpichler** [comment]
All checks are ok now - please merge

**Review by NL66278 — APPROVED** [review]
:+1: LGTM

**Pull request merged**